packages/block-library/src/list/v2/migrate.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+
+function createListBlockFromDOMElement( listElement ) {
+	const listAttributes = {
+		ordered: 'OL' === listElement.tagName,
+		start: listElement.getAttribute( 'start' )
+			? parseInt( listElement.getAttribute( 'start' ), 10 )
+			: undefined,
+		reversed:
+			listElement.getAttribute( 'reversed' ) === true ? true : undefined,
+		type: listElement.getAttribute( 'type' ) ?? undefined,
+	};
+
+	const innerBlocks = Array.from( listElement.children ).map(
+		( listItem ) => {
+			const children = Array.from( listItem.childNodes );
```

Comment:
Assuming we want to use createListBlockFromDOMElement as the function that does the rawHandling  of HTML to list blocks. Would something like calling the rawHandler on the innerHTML of a list node and then making the first paragraph content a list-item work
Something like:
```
 [ firstParagraph, innerBlocks ] =
					rawHandler( {
						HTML: listItem.innerHTML,
						mode: 'BLOCKS',
					} );

			return createBlock(
				'core/list-item',
				{childAttributes, content: firstParagraph.content,
				innerBlocks
			);
```

If in the future we allow other child blocks inside the list item this solution seems future-proof, and right now also handles newline text nodes, etc.

Reply:
We don't need this function to be future proof, this is just for the old markup migration like said in the previous comment. If there's random markup there, the block was probably invalid anyway.